### PR TITLE
feat(CardDAV): Add Sabre\DAV\IMoveTarget support to OCA\DAV\CardDAV\AddressBook

### PR DESCRIPTION
**Pull request by tcitworld**
## Summary

This allows to just `UPDATE` the card row instead of deleting it and reinsert it. It's very similar to https://github.com/nextcloud/server/pull/30120 for calendars. It's in a transaction as well since there are multiple operations.

As we need the `addressbookid` exposed, this introduces `OCA\DAV\CardDAV\Card` that extends Sabre's.

I chose specifically NOT to auto-inject `LoggerInterface` in `Addressbook` like in #30120 because the chain of DI is huge just for ONE simple call and it would break an existing dirty call (`OCA\Contacts` calling `OCA\DAV`) of `ContactsManager` in Contacts: https://github.com/nextcloud/contacts/pull/1722 (in `SocialApiService`), but this is debatable (and we can prioritize making the public interfaces correct first).

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by miaulalala — APPROVED** [review]

**Comment by miaulalala** [comment]
Psalm isn't happy

```
Error: 3rdparty/sabre/dav/lib/CardDAV/Card.php:85:25: MoreSpecificImplementedParamType: Argument 1 of  
 Sabre\CardDAV\Card::put has the more specific type 'string', expecting 'resource|string' as defined by  
 Sabre\DAV\IFile::put (see https://psalm.dev/140)
```

Drone failure is contacts-menu as usual

**Comment by tcitworld** [comment]
For the psalm issue, I guess I need to update the baseline to add this. Done.

For the contacts menu test failure, this time it *might* be related. Does it fails elsewhere currently?

File: apps/dav/lib/CardDAV/Card.php [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program. If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+declare(strict_types=1);
```
Comment (come-nc) [inline apps/dav/lib/CardDAV/Card.php:22]: Move strict_types before copyright for consistency accross files.

**Review by come-nc — APPROVED** [review]

**Comment by szaimen** [comment]
conflicts

**Comment by szaimen** [comment]
CI failure unrelated afaics

**Pull request merged**